Lab01_Exercise01/exercise01.c

```diff
@@ -1,39 +1,60 @@
 #include <stdio.h>
 
-
+// Ex 1.5 `#include` statements, cause the specified file to be recursively copied into the source file at the location of the statement prior to compilation
+// For this reason it's normal for headers to use macros to prevent them from being included multiple times in the header hierarchy.
 #include "random.h"
 
+// Ex 1.1, Pre-processor definitions (or Macros) take the form `#define <name>` where the remainder of the line provides the replacement
+// The compiler's pre-processor, will replace occurrences of `NUM_VALUES` with `250` in the source prior to compilation
 #define NUM_VALUES 250
+// Ex 1.2, Arrays declared in this form must have a known size at compile time
+// In this case, the compiler will know the length to be 250, due to the replacement of `NUM_VALUES`
 int values [NUM_VALUES];
 
 int main()
 {
-
-	unsigned int sum = 0;
-	unsigned int average = 0;
-	int min = 0;
-	int max = 0;
-	unsigned char i = 0;	
-
-	init_random();
-
-	for (i=0; i<NUM_VALUES; i++){
-		values[i] = random_ushort();
-		
-		sum += values[i];
-	}
-
-	average = sum / NUM_VALUES;
-
-	for (i=0; i<NUM_VALUES; i++){
-		values[i] -= average;
-		min = (values[i] < min)? values[i] : min;
-		max = (values[i] > max)? values[i]: max;
-	}
-
-	printf ("Sum is %u\n", sum);
-	printf ("Average is %u\n", average);
-	printf ("Min is %d\n", min);
-	printf ("Max is %d\n", max);
+    // Ex 1.3, sum is declared, local to the function main(), and initialised to 0
+    // As an unsigned integer, it cannot hold negative values, but has a greater positive range than a signed integer
+    unsigned int sum = 0;
+    // Ex 1.8 (2/3), an unsigned int has [0, 2^32) range, it should be suitable to hold the sum of 250 unsigned shorts
```

Comment:
```suggestion
    // Ex 1.8 (2/3), an unsigned int has [0, 2^32] range, it should be suitable to hold the sum of 250 unsigned shorts
```

Reply:
Range of uint is from 0 to 2^32 - 1, hence the inclusive-exclusive `[)` range notation.